tests/core/test_core_unique_ptr.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "vecmem/memory/host_memory_resource.hpp"
+#include "vecmem/memory/memory_resource.hpp"
+#include "vecmem/memory/unique_ptr.hpp"
+
+namespace {
+static vecmem::host_memory_resource mr;
```

Comment:
Why is the MSVC error nonsense? In that context the `mr` name hides the `mr` variable declared here.

You can argue that we shouldn't care about such symbol name hiding in the first place. But as long as that is enabled as a warning, MSVC is correctly identifying that that is what's happening here... 🤔

Reply:
It's identifying the shadowing correctly, but think about what the consequence of this is: you would be completely unable to ever use any global variables because you would run the risk of importing some header that uses the same name in a function somewhere... That is totally unworkable. Sure, you could argue that you should not have global variables in the first place, but when writing tests they are pretty useful...

Reply:
:confused:  We are building VecMem with a higher-than-default warning level.

https://github.com/acts-project/vecmem/blob/main/cmake/vecmem-compiler-options-cpp.cmake#L32

We've already come across this exact behaviour in #93. As we discussed back then, template instantiations work in curious ways in MSVC. Also as we discussed there, we should just respect this behaviour of MSVC at `/W4`. Client code of VecMem doesn't have to do so though.

So just rename the memory resource in the test (`mr` is a bad name for a global variable anyway), and let's be done with this...

Reply:
Done. :slightly_smiling_face: